lib/runner.js

```diff
@@ -57,29 +64,49 @@ class Runner {
         })
 
         global.browser.on('command', (payload) => {
-            process.send({
+            const command = {
                 event: 'runner:command',
                 cid: m.cid,
+                specs: this.specs,
                 method: payload.method,
                 uri: payload.uri,
                 data: payload.data
-            })
+            }
+            process.send(this.addTestDetails(command))
         })
 
         global.browser.on('result', (payload) => {
-            process.send({
+            const result = {
                 event: 'runner:result',
                 cid: m.cid,
+                specs: this.specs,
                 requestData: payload.requestData,
                 requestOptions: payload.requestOptions,
                 body: payload.body // ToDo figure out if this slows down the execution time
-            })
+            }
+            process.send(this.addTestDetails(result))
+        })
+
+        global.browser.on('screenshot', (payload) => {
+            const details = {
+                event: 'runner:screenshot',
+                cid: m.cid,
+                specs: this.specs,
+                filename: payload.filename,
+                data: payload.data
+            }
+            process.send(this.addTestDetails(details))
+        })
+
+        process.on('test:start', (test) => {
```

Comment:
See [comment](https://github.com/webdriverio/webdriverio/pull/896/files#r47439165). Can't we move this to the basereporter? and apply it to the test data there?